Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/child_app/view/task_list/TaskRecyclerViewAdapter.java

```diff
@@ -17,29 +17,53 @@
     private TaskItemClickListener taskItemClickListener;
 
     protected interface TaskItemClickListener {
+        void timerIconListener(int position);
         void stepsIconListener(int position);
     }
 
     static class TaskRecyclerViewHolder extends RecyclerView.ViewHolder {
         TextView taskName;
-        ImageView stepsPicture;
+        TextView taskDuration;
+        ImageView actionPicture;
         TaskItemClickListener taskItemClickListener;
+        boolean hasTimer = false;
 
         TaskRecyclerViewHolder(View itemView, final TaskItemClickListener taskItemClickListener) {
             super(itemView);
             this.taskName = (TextView) itemView.findViewById(R.id.id_tv_task_name);
-            this.stepsPicture = (ImageView) itemView.findViewById(R.id.id_step_list_icon);
-            this.stepsPicture.setOnClickListener(stepsIconListener);
+            this.actionPicture = (ImageView) itemView.findViewById(R.id.id_task_activity_icon);
+            this.taskDuration = (TextView) itemView.findViewById(R.id.id_tv_task_duration_time);
+            this.actionPicture.setOnClickListener(actionIconListener);
             this.taskItemClickListener = taskItemClickListener;
         }
 
-
-        View.OnClickListener stepsIconListener = new View.OnClickListener() {
+        View.OnClickListener actionIconListener = new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                taskItemClickListener.stepsIconListener(getAdapterPosition());
+                if(hasTimer)
+                    taskItemClickListener.timerIconListener(getAdapterPosition());
+                else
+                    taskItemClickListener.stepsIconListener(getAdapterPosition());
             }
         };
+
+        protected void setUpHolder(TaskTemplate task){
+            taskName.setText(task.getName());
+
+            if (!task.getStepTemplates().isEmpty()) {
+                actionPicture.setImageResource(R.drawable.ksiazki);
+                taskDuration.setVisibility(View.INVISIBLE);
+            }
+            else if (task.getDurationTime() != null) {
+                actionPicture.setImageResource(R.drawable.timer);
+                taskDuration.setText(String.format("%ss", task.getDurationTime().toString()));
+                hasTimer = true;
+            }
+            else {
+                actionPicture.setVisibility(View.INVISIBLE);
+                taskDuration.setVisibility(View.INVISIBLE);
+            }
+        }
     }
 
     TaskRecyclerViewAdapter(List<TaskTemplate> tasks, TaskItemClickListener taskItemClickListener) {
```

Comment:
There are two ClickListeners for task item right now. For certain we need StepsClickListener to be passed here, not sure about about TimerClickListener. So I would recommend to refactor taskItemClickListener to StepsIconListener/StepsClickListener, to make code more clear.